cunumeric/_ufunc/ufunc.py

```diff
@@ -753,6 +753,13 @@ def reduce(
                 f"reduction for {self} is not yet implemented"
             )
 
+        if self._op_code in [
```

Comment:
This might not be necessary if you use `red_code=UnaryRedCode.ALL`, since the implementation of that reduction operator converts all datatypes to bools "just-in-time" before combining them with `AND`. https://github.com/nv-legate/cunumeric/blob/branch-24.01/src/cunumeric/unary/unary_red_util.h#L126

Reply:
When I try this, I get an error saying

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Type mismatch: bool accessor to a float64 store. Disable type checking via accessor template parameter if this is intended.

Reply:
OK, the problem is that we need to feed `res_dtype=bool` to `_perform_unary_reduction` when we want the result array to have boolean type irrespective of the type of the argument array. So let's do that for `BinaryOpCode.LOGICAL_AND` and `BinaryOpCode.LOGICAL_OR`, and also set `UnaryRedCode.ALL`/`UnaryRedCode.ANY` as the `red_code` for these ufuncs.

Also of note, `dtype` and `res_dtype` are mutually incompatible, so we will need to raise a `TypeError` if `dtype` is set and we're dealing with one of these two ufuncs.

The reason I'm pushing for this is that `_astype` creates a temporary array, which was can easily avoid.

Reply:
I see what you mean now, and that does work. Thank you. I hadn't made the connection between ALL as an AND and ANY as an OR. However, I have two slight problems. At present, logical_xor.reduce._cunumeric.implemented actually returns True as it seems the only implementation check comes from within the .reduce() function itself where it checks the binary_ufunc's red_code. Also do you know why the build in the tests is failing? Intuitively, I don't think I made any changes that would affect the build itself, and the error messages seem a little unrelated.

Reply:
> AAt present, logical_xor.reduce._cunumeric.implemented actually returns True as it seems the only implementation check comes from within the .reduce() function itself where it checks the binary_ufunc's red_code.

Left a comment on the thread for this.

> Also do you know why the build in the tests is failing? Intuitively, I don't think I made any changes that would affect the build itself, and the error messages seem a little unrelated.

Yes, looks like a fix has been merged onto the main branch in the meantime. I merged it into this branch, hopefully all should work now.